src/mirrors.rs

```diff
@@ -0,0 +1,118 @@
+/// Implementation of the PostgreSQL server (database) protocol.
+/// Here we are pretending to the a Postgres client.
+use bytes::{BytesMut};
+
+use parking_lot::Mutex;
+use tokio::sync::mpsc::{Receiver, Sender, channel};
+
+use crate::config::{Address, User};
+use crate::pool::ClientServerMap;
+use crate::server::Server;
+use crate::stats::get_reporter;
+
+pub enum MirrorOperation {
+    Send(BytesMut),
+    Receive
+}
+pub struct MirrorUnit {
+    pub address: Address,
+    pub user: User,
+    pub database: String,
+    pub bytes_rx: Receiver<MirrorOperation>,
+    pub exit_rx: Receiver<()>,
+}
+
+impl MirrorUnit {
+    pub fn begin(mut self, server_id: i32) {
+        tokio::spawn(async move {
+            let mut server = Server::startup(
+                server_id,
+                &self.address.clone(),
+                &self.user.clone(),
+                self.database.as_str(),
+                ClientServerMap::default(),
+                get_reporter()
+            ).await.unwrap();
+
+            loop {
+                tokio::select! {
+                    _ = self.exit_rx.recv() => {
+                        break;
+                    }
+                    op = self.bytes_rx.recv() => {
+                        match op {
+                            Some(MirrorOperation::Send(bytes)) => {
+                                server.send(&bytes).await.unwrap();
```

Comment:
We'll probably want to catch errors here and report errors or reconnect. 
I worry about reconnections being too aggressive.